container/push.bzl

```diff
@@ -47,6 +47,12 @@ def _impl(ctx):
 
     # Parse and get destination registry to be pushed to
     registry = ctx.expand_make_variables("registry", ctx.attr.registry, {})
+
+    # If a registry file is provided, override <registry> with file value
+    if ctx.file.registry_file:
```

Comment:
To make it clearer that this is overriding registry could we do something like:

```
if not ctx.file.registry_file:
  registry = ctx.expand_make_variables("registry", ctx.attr.registry, {})
else:
   registry = "$(cat {})"....
```

I wasn't immediately able to see that this was overriding this field (I had assumed `pusher_input.append` was adding an argument or something).

Reply:
I'm not sure on shifting this one, as the pattern that the file has follows the "assignment, then if override" ([repository_file](https://github.com/bazelbuild/rules_docker/blob/74be3091ffc0d2fc4d4050d9524519e848d0868c/container/push.bzl#L53), [tag_file](https://github.com/bazelbuild/rules_docker/blob/74be3091ffc0d2fc4d4050d9524519e848d0868c/container/push.bzl#L60), [client_config](https://github.com/bazelbuild/rules_docker/blob/74be3091ffc0d2fc4d4050d9524519e848d0868c/container/push.bzl#L108)) pattern.

I am fine with changing it, but it would break with the pattern in the method.